### ISSUE TITLE
[BUG] Plots.jl v1.11.0+ error in Julia 1.6: ERROR: ArgumentError: reducing over an empty collection is not allowed

### ISSUE THREAD

**Issue by truedichotomy** [JuliaPlots/Plots.jl#3381]
## Details
The following code runs fine in Julia v1.5 but not v1.6 when using Plots.jl v1.11.0. Prior to this version of Plotly.jl, the code will run in both Julia v1.5 and v1.6:

```julia
a = collect(1:10)
b = Array{Float64}(undef,10)
b .= NaN
Plots.plot(a,b)
```

Here is the error output in Julia v1.6:
```julia
julia> Plots.plot(a,b)
Error showing value of type Plots.Plot{Plots.GRBackend}:
ERROR: ArgumentError: reducing over an empty collection is not allowed
Stacktrace:
  [1] _empty_reduce_error()
    @ Base ./reduce.jl:299
  [2] mapreduce_empty(f::Function, op::Base.MappingRF{Plots.var"#7#14", Base.BottomRF{typeof(min)}}, T::Type)
    @ Base ./reduce.jl:342
  [3] reduce_empty(op::Base.MappingRF{Plots.var"#6#13", Base.MappingRF{Plots.var"#7#14", Base.BottomRF{typeof(min)}}}, #unused#::Type{Any})
    @ Base ./reduce.jl:329
  [4] reduce_empty_iter
    @ ./reduce.jl:355 [inlined]
  [5] reduce_empty_iter
    @ ./reduce.jl:354 [inlined]
  [6] foldl_impl(op::Base.MappingRF{Plots.var"#6#13", Base.MappingRF{Plots.var"#7#14", Base.BottomRF{typeof(min)}}}, nt::Base._InitialValue, itr::Vector{Any})
    @ Base ./reduce.jl:49
  [7] mapfoldl_impl(f::typeof(identity), op::typeof(min), nt::Base._InitialValue, itr::Base.Generator{Base.Generator{Vector{Any}, Plots.var"#6#13"}, Plots.var"#7#14"})
    @ Base ./reduce.jl:44
  [8] #mapfoldl#214
    @ ./reduce.jl:160 [inlined]
  [9] mapfoldl
    @ ./reduce.jl:160 [inlined]
 [10] #mapreduce#218
    @ ./reduce.jl:287 [inlined]
 [11] mapreduce
    @ ./reduce.jl:287 [inlined]
 [12] #minimum#229
    @ ./reduce.jl:763 [inlined]
 [13] minimum(a::Base.Generator{Base.Generator{Vector{Any}, Plots.var"#6#13"}, Plots.var"#7#14"})
    @ Base ./reduce.jl:763
 [14] series_segments(series::Plots.Series, seriestype::Symbol)
    @ Plots ~/.julia/packages/Plots/z5Msu/src/utils.jl:97
 [15] gr_draw_segments(series::Plots.Series, x::Vector{Float64}, y::Vector{Float64}, fillrange::Nothing, clims::Tuple{Float64, Float64})
    @ Plots ~/.julia/packages/Plots/z5Msu/src/backends/gr.jl:1670
 [16] gr_add_series(sp::Plots.Subplot{Plots.GRBackend}, series::Plots.Series)
    @ Plots ~/.julia/packages/Plots/z5Msu/src/backends/gr.jl:1613
 [17] gr_display(sp::Plots.Subplot{Plots.GRBackend}, w::Measures.AbsoluteLength, h::Measures.AbsoluteLength, viewport_canvas::Vector{Float64})
    @ Plots ~/.julia/packages/Plots/z5Msu/src/backends/gr.jl:935
 [18] gr_display(plt::Plots.Plot{Plots.GRBackend}, fmt::String)
    @ Plots ~/.julia/packages/Plots/z5Msu/src/backends/gr.jl:626
 [19] gr_display
    @ ~/.julia/packages/Plots/z5Msu/src/backends/gr.jl:593 [inlined]
 [20] _display(plt::Plots.Plot{Plots.GRBackend})
    @ Plots ~/.julia/packages/Plots/z5Msu/src/backends/gr.jl:1895
 [21] display(#unused#::Plots.PlotsDisplay, plt::Plots.Plot{Plots.GRBackend})
    @ Plots ~/.julia/packages/Plots/z5Msu/src/output.jl:150
 [22] display(x::Any)
    @ Base.Multimedia ./multimedia.jl:328
 [23] #invokelatest#2
    @ ./essentials.jl:708 [inlined]
 [24] invokelatest
    @ ./essentials.jl:706 [inlined]
 [25] print_response(errio::IO, response::Any, show_value::Bool, have_color::Bool, specialdisplay::Union{Nothing, AbstractDisplay})
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:247
 [26] (::REPL.var"#40#41"{REPL.LineEditREPL, Pair{Any, Bool}, Bool, Bool})(io::Any)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:231
 [27] with_repl_linfo(f::Any, repl::REPL.LineEditREPL)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:462
 [28] print_response(repl::REPL.AbstractREPL, response::Any, show_value::Bool, have_color::Bool)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:229
 [29] (::REPL.var"#do_respond#61"{Bool, Bool, REPL.var"#72#82"{REPL.LineEditREPL, REPL.REPLHistoryProvider}, REPL.LineEditREPL, REPL.LineEdit.Prompt})(s::REPL.LineEdit.MIState, buf::Any, ok::Bool)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:798
 [30] (::REPL.var"#77#87")(::REPL.LineEdit.MIState, ::Any, ::Vararg{Any, N} where N)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:1068
 [31] #invokelatest#2
    @ ./essentials.jl:708 [inlined]
 [32] invokelatest
    @ ./essentials.jl:706 [inlined]
 [33] (::REPL.LineEdit.var"#22#23"{REPL.var"#77#87", String})(s::Any, p::Any)
    @ REPL.LineEdit /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/LineEdit.jl:1414
 [34] prompt!(term::REPL.Terminals.TextTerminal, prompt::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/LineEdit.jl:2534
 [35] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/LineEdit.jl:2436
 [36] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:1126
 [37] (::REPL.var"#44#49"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL ./task.jl:406
```

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |   x  |     |
pyplot       |  x   |     |
plotly       |   x  |     |
plotlyjs     |  x  |    |
pgfplotsx    |     |     | x
inspectdr    |     |     | x

### Versions

Plots.jl version:
Backend  version (`]st -m`):
Plotly v0.3.0
Plots v1.11.0

Output of `versioninfo()`:
Julia Version 1.6.0
Commit f9720dc2eb (2021-03-24 12:55 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin19.6.0)
  CPU: Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 4

**Comment by truedichotomy** [comment]
If just one of the element in y is change to a non-NaN value, the plotting code above will work. I hope this bug can be addressed as I generate a lot of figures programmatically and right now I have to manually work around this bug. Thanks.

**Comment by truedichotomy** [comment]
NaN is a convenient placeholder for plots that have `yscale=:log10`, because that log10 flag does not like 0's. I use NaN to represent 0 when plotting in log scale. Otherwise Plots give me this error:

```
ERROR: ArgumentError: At least one finite value must be provided to formatter.
Stacktrace:
  [1] showoff(xs::Vector{Float64}, style::Symbol)
    @ Showoff ~/.julia/packages/Showoff/C1Bj4/src/Showoff.jl:111
  [2] optimal_ticks_and_labels(sp::Plots.Subplot{Plots.PlotlyBackend}, axis::Plots.Axis, ticks::Nothing)
    @ Plots ~/.julia/packages/Plots/SjqWU/src/axes.jl:199
  [3] optimal_ticks_and_labels
    @ ~/.julia/packages/Plots/SjqWU/src/axes.jl:141 [inlined]
  [4] get_ticks(sp::Plots.Subplot{Plots.PlotlyBackend}, axis::Plots.Axis; update::Bool)
    @ Plots ~/.julia/packages/Plots/SjqWU/src/axes.jl:248
  [5] get_ticks
    @ ~/.julia/packages/Plots/SjqWU/src/axes.jl:223 [inlined]
  [6] tick_padding(sp::Plots.Subplot{Plots.PlotlyBackend}, axis::Plots.Axis)
    @ Plots ~/.julia/packages/Plots/SjqWU/src/backends.jl:76
  [7] _update_min_padding!(sp::Plots.Subplot{Plots.PlotlyBackend})
    @ Plots ~/.julia/packages/Plots/SjqWU/src/backends.jl:106
  [8] iterate
    @ ./generator.jl:47 [inlined]
  [9] _collect(c::Matrix{RecipesBase.AbstractLayout}, itr::Base.Generator{Matrix{RecipesBase.AbstractLayout}, typeof(Plots._update_min_padding!)}, #unused#::Base.EltypeUnknown, isz::Base.HasShape{2})
    @ Base ./array.jl:691
 [10] collect_similar
    @ ./array.jl:606 [inlined]
 [11] map
    @ ./abstractarray.jl:2294 [inlined]
 [12] _update_min_padding!(layout::Plots.GridLayout)
    @ Plots ~/.julia/packages/Plots/SjqWU/src/layouts.jl:282
 [13] iterate
    @ ./generator.jl:47 [inlined]
 [14] _collect(c::Matrix{RecipesBase.AbstractLayout}, itr::Base.Generator{Matrix{RecipesBase.AbstractLayout}, typeof(Plots._update_min_padding!)}, #unused#::Base.EltypeUnknown, isz::Base.HasShape{2})
    @ Base ./array.jl:691
 [15] collect_similar
    @ ./array.jl:606 [inlined]
 [16] map
    @ ./abstractarray.jl:2294 [inlined]
 [17] _update_min_padding!(layout::Plots.GridLayout)
    @ Plots ~/.julia/packages/Plots/SjqWU/src/layouts.jl:282
 [18] prepare_output(plt::Plots.Plot{Plots.PlotlyBackend})
    @ Plots ~/.julia/packages/Plots/SjqWU/src/plot.jl:189
 [19] display(#unused#::Plots.PlotsDisplay, plt::Plots.Plot{Plots.PlotlyBackend})
    @ Plots ~/.julia/packages/Plots/SjqWU/src/output.jl:149
 [20] display(x::Any)
    @ Base.Multimedia ./multimedia.jl:328
 [21] #invokelatest#2
    @ ./essentials.jl:708 [inlined]
 [22] invokelatest
    @ ./essentials.jl:706 [inlined]
 [23] print_response(errio::IO, response::Any, show_value::Bool, have_color::Bool, specialdisplay::Union{Nothing, AbstractDisplay})
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:247
 [24] (::REPL.var"#40#41"{REPL.LineEditREPL, Pair{Any, Bool}, Bool, Bool})(io::Any)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:231
 [25] with_repl_linfo(f::Any, repl::REPL.LineEditREPL)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:462
 [26] print_response(repl::REPL.AbstractREPL, response::Any, show_value::Bool, have_color::Bool)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:229
 [27] (::REPL.var"#do_respond#61"{Bool, Bool, REPL.var"#72#82"{REPL.LineEditREPL, REPL.REPLHistoryProvider}, REPL.LineEditREPL, REPL.LineEdit.Prompt})(s::REPL.LineEdit.MIState, buf::Any, ok::Bool)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:798
 [28] (::REPL.var"#77#87")(::REPL.LineEdit.MIState, ::Any, ::Vararg{Any, N} where N)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:1068
 [29] #invokelatest#2
    @ ./essentials.jl:708 [inlined]
 [30] invokelatest
    @ ./essentials.jl:706 [inlined]
 [31] (::REPL.LineEdit.var"#22#23"{REPL.var"#77#87", String})(s::Any, p::Any)
    @ REPL.LineEdit /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/LineEdit.jl:1414
 [32] prompt!(term::REPL.Terminals.TextTerminal, prompt::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/LineEdit.jl:2534
 [33] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/LineEdit.jl:2436
 [34] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:1126
 [35] (::REPL.var"#44#49"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL ./task.jl:406
```

**Comment by jmarbet** [comment]
I have the same issue and hope this can be addressed. There seems to be another issue related to this: #3372

Did you find a good workaround (besides setting one element of the vector to some number)?

**Comment by yha** [comment]
Fixed by #3468
